_rules/links-identical-name-equivalent-purpose-b20e66.md

```diff
@@ -24,24 +25,30 @@ htmlHintIgnore:
 
 ## Applicability
 
-This rule applies to any set of any two or more HTML or SVG elements that have the [semantic role](#semantic-role) of `link`, or a role that inherits from the `link` role, are [included in the accessibility tree](#included-in-the-accessibility-tree), and that have [matching](#matching-characters) [accessible names][accessible name] that do not only consist of [whitespace](#whitespace).
+This rule applies to any set of any two or more HTML or SVG elements which
 
-**Note:** The test target for this rule is the full set of link elements that share the same [matching](#matching-characters) [accessible name][].
+- have the [semantic role][] of `link`, or a role that inherits from the `link` role; and
+- are in [documents][document] of the same [web page][]; and
+- are [included in an accessibility tree][included in the accessibility tree]; and
+- that have [matching][] [accessible names][accessible name] that do not only consist of [whitespace](#whitespace).
+
+**Note:** The test target for this rule is the full set of link elements that share the same [matching][] [accessible name][].
 
 ## Expectation
 
-When followed, the links in each set of target elements resolve to the [same resource](#same-resource) or to [equivalent resources](#equivalent-resource). Resolving the links includes potential redirects, if the redirects happen instantly.
+When followed, the links in each set of target elements resolve to the [same resource][] or to [equivalent resources](#equivalent-resource). Resolving the links includes potential redirects, if the redirects happen instantly.
```

Comment:
According to SC 2.4.9 the purpose of each link can be determined from the link text alone. I do not think the test rule tests this.

According to the test rule, links with matching names should resolve to the same or equivalent resource. In my opinion, this is best practice, but not a requirement.

Pasted from the SC 2.4.9 understanding article: "Best practice is that links with the same destination would have the same descriptions, but links with different purposes and destinations would have different descriptions (see also Success Criterion 3.2.4 which calls for consistency in identifying components that have the same functionality)." https://www.w3.org/WAI/WCAG21/Understanding/link-purpose-link-only.html 

Also, if the linked resource(s) with matching accessible names have a different purpose than what is described in the link name, this would pass the test rule, but fail the SC.

In my opinion, this test rule is more relevant for SC 3.2.4 Consistent Identification. Either way, SC 3.2.4 requires the identification to be consistent, but not necessarily identical.

I have also left a similar comment, related to SC 2.4.4 in #693

Reply:
…Hmmmm…

---
I agree with the fact that it is more a best practice than a failure for [2.4.9](https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-link.html). However, I am not sure this is a failure for [3.2.4](https://www.w3.org/TR/UNDERSTANDING-WCAG20/consistent-behavior-consistent-functionality.html) either because of this:
> While it is desirable and best practice always to be consistent within a single web page, 3.2.4 only addresses consistency within a set of web pages where something is repeated on more than one page in the set.

---
Also, the problem is present in [the current rule](https://act-rules.github.io/rules/b20e66) and not really related to this PR. It certainly need to be fixed but this might require more discussion and a separate PR to keep things focused.

---
My current approach would be to keep this PR as is (not changing the SC), and open an issue for the SC mismatch that can be discussed further and fixed later. Would that be OK for you @DagfinnRomen ?

Reply:
That would be ok, @Jym77 . Could you open an issue for this?

Reply:
Will do.

Reply:
This is definitely something for a separate pull request, yes. My opinion on the matter is that the rule is really making an assumption that has not yet been written out. That assumption is that when two links have the same accessible name, they describe the same purpose, and the resources they reference should therefore be equivalent as they otherwise do not serve the same purpose. 

Put another way, the word "coffee" is not really describing the purpose of "a cup of coffee" if I also use the word to refer to "a cup of tea". In reverse, I should only use the word "coffee", and not also "joe", "java", or "mocha", when I want to refer to "a cup of coffee"; this seems to be the best practice described in the understanding document.

Reply:
> Pasted from the SC 2.4.9 understanding article: "Best practice is that links with the same destination would have the same descriptions, but links with different purposes and destinations would have different descriptions (see also Success Criterion 3.2.4 which calls for consistency in identifying components that have the same functionality)." https://www.w3.org/WAI/WCAG21/Understanding/link-purpose-link-only.html

Hmmmm…
This seems to go the other way around. (and that is likely what @kasperisager is also saying…)

The best practice is that if the links go to the same place, then they should have the same name. The rule is that if the links have the same name, then they should lead to the same place.

So, to keep on with @kasperisager examples:
```html
<a href="#cup-of-coffee">coffee</a>
<a href="#cup-of-coffee">java</a>
```
is failing the best practice (both links have same destination but different names), but not the SC. And it is inapplicable for the rule (there is no set of two or more links with the same name).

On the other hand,
```html
<a href="#cup-of-coffee">coffee</a>
<a href="#cup-of-tea">coffee</a>
```
is not failing the best practice (the links do not have the same destination), but is failing the rule (they have same name but different destination), and likely failing the SC (the link text alone is not enough to get the purpose of the link given that the same text cannot accurately describes two different purposes).

---
Therefore, I am now thinking that the accessibility requirement mapping is actually correct. What do you think @DagfinnRomen ?

Reply:
I am with you so far, @Jym77 and @kasperisager . But what about the following?
```
<a href="#cup-of-coffee">tea</a>
<a href="#cup-of-coffee">tea</a>
```
This passes both the best practice and the rule (the links are identical and have identical destinations). However, it fails the SC, as the link texts do not describe the purpose.

Reply:
Yes.

As always, the rule is only a partial check and only has a fail-on-fail relationship with the SC. The rule can pass, but the SC still fail, hence the "All passed outcomes: further testing is needed" requirement mapping.

Locally, (for the "in context" version), it would fail #555 (I do not think we have a similar rule without context given that 2.4.9 is AAA, but it would probably be super easy to have by just taking #555 and removing context…)

Reply:
@DagfinnRomen I made a separate PR #959 to improve one inapplicable example by having the "same link, different names" case.

Resolving this conversation.

Reply:
I am not sure if the test rule has a fail-on-fail relationship. The failed test cases are all based on the assumption that the same text cannot accurately describe two different purposes.

This may be considered best practice, but it is not a part of the requirement.

For example
```
<a href="#cup-of-coffee">coffee</a>
<a href="#cup-of-joe">coffee</a>
```
This fails both best practice and the rule, but it does not fail the SC.

The requirement states that each link describes the purpose of the link. In my opinion, the links are to be assessed individually. It is not necessary to compare links to verify SC 2.4.9.

I support the suggestion of adapting #555 to a new rule for SC 2.4.9, though.

Reply:
While the links are not **identical**, they are likely [**equivalent**](https://act-rules.github.io/glossary/#equivalent-resource) and thus the rule passes.

On the other hand, if `#cup-of-joe` is about the personal cup of a guy named Joe (and `#cup-of-coffee` is about coffee), then the links are not equivalents and the rule fails, but so does the SC (although for a different reason, namely that `"coffee"` is not an accurate description of the cup belonging to Joe).

So, this should be good 🤞 

---
I do agree, however, that the SC is about individual links rather than comparing links.
I do not know why this rule was here in the first place.
We may actually want to kill it once we have #555 (and its "no context" version) since if each individual link is descriptive, then links with same name must lead to equivalent resources.

This may require further discussion (and if we consider that the rule has been superseded and we plan to kill it, there's no need for this PR 🙄 )

Reply:
@DagfinnRomen, check also #962.

Reply:
As a conclusion to #968, an assumption is added to this rule in #980.
This should fix this concern.